src/prefect/environments/execution/cloud/environment.py

```diff
@@ -0,0 +1,151 @@
+import json
+import logging
+import time
+import uuid
+from os import path
+from typing import List
+
+import docker
+import yaml
+
+import prefect
+from prefect.environments.storage import Docker
+from prefect.environments.execution import Environment
+
+
+class CloudEnvironment(Environment):
+    """"""
+
+    def __init__(self) -> None:
+        """"""
+        self.identifier_label = str(uuid.uuid4())
+
+    def process(self, storage: "Docker" = Docker()) -> None:
+        if type(storage) != prefect.environments.storage.Docker:
+            raise TypeError("CloudEnvironment requires a Docker storage option")
+
+        if not storage.image_name or not storage.image_tag or not storage.registry_url:
+            raise ValueError("Docker storage is missing required fields")
+
+        self.execute(
+            registry_url=storage.registry_url,
+            image_name=storage.image_name,
+            image_tag=storage.image_tag,
+        )
+
+    def execute(self, registry_url: str, image_name: str, image_tag: str) -> None:
+        """"""
+        from kubernetes import client, config
+
+        try:
+            config.load_incluster_config()
+        except config.config_exception.ConfigException:
+            raise EnvironmentError("Environment not currently inside a cluster")
+
+        batch_client = client.BatchV1Api()
+
+        with open(path.join(path.dirname(__file__), "job.yaml")) as job_file:
+            job = yaml.safe_load(job_file)
+            job = self._populate_job_yaml(
+                yaml_obj=job,
+                registry_url=registry_url,
+                image_name=image_name,
+                image_tag=image_tag,
+            )
+
+            # Create Job
+            batch_client.create_namespaced_job(namespace="default", body=job)
+
+    def run(
+        self, flow_file_path: str = "/root/.prefect/flow_env.prefect"
+    ) -> "prefect.engine.state.State":
+        """"""
+        from prefect.engine.executors import DaskExecutor
+        from dask_kubernetes import KubeCluster
+
+        with open(path.join(path.dirname(__file__), "worker_pod.yaml")) as pod_file:
+            worker_pod = yaml.safe_load(pod_file)
+            worker_pod = self._populate_worker_pod_yaml(yaml_obj=worker_pod)
+
+            cluster = KubeCluster.from_dict(worker_pod)
+            cluster.adapt(minimum=1, maximum=1)
+
+            schema = prefect.serialization.flow.FlowSchema()
+            with open(flow_file_path, "r") as f:
+                flow = schema.load(json.load(f))
+
+                executor = DaskExecutor(address=cluster.scheduler_address)
+                flow.run(executor=executor)
```

Comment:
We shouldn't use `flow.run` in environments

Reply:
Which method would you suggest?

Reply:
Using a `FlowRunner` like:
```python
FlowRunner(flow=flow).run(executor=executor)
```